xayn-ai/src/coi/config.rs

```diff
@@ -7,11 +9,18 @@ pub(crate) struct Configuration {
     pub threshold: f32,
     /// The positive number of neighbors for the k-nearest-neighbors distance.
     pub neighbors: NonZeroUsize,
-    /// The maximum number of key phrases picked during the coi key phrase selection. A coi may have
-    /// more key phrases than this, eg because of merging.
-    pub max_key_phrases: usize,
+    /// The time since the last view after which a coi becomes irrelevant.
+    pub horizon: Duration,
     /// The weighting between coi and pairwise candidate similarites in the key phrase selection.
     pub gamma: f32,
+    /// The penalty for less relevant key phrases in the top key phrase selection.
+    pub penalty: [f32; Self::MAX_KEY_PHRASES],
+}
+
+impl Configuration {
+    /// The maximum number of key phrases picked during the coi key phrase selection. A coi may have
+    /// more key phrases than this, eg because of merging.
+    pub const MAX_KEY_PHRASES: usize = 3;
```

Comment:
Can't we then use `penalty.len()` as `max_key_phrases`? This with `penalty: [f32; Self::MAX_KEY_PHRASES]` constraints the configuration more then needed.